File: InternetForumFrontend/src/app/Models/PostCommentVote.ts
import { PostComment } from "./PostComment";
import { UserPostCommentVote } from "./UserPostCommentVote";


// NOTE: ? = null, ! = not null
export class PostCommentVote{ 
        voteId?: number;
        voteCount!: number;
        postCommentId?: number;
        postComment?: PostComment;
        userPostCommentVotes?: UserPostCommentVote;
}

File: InternetForumBackend/API/Data/Models/PostCommentVote.cs
namespace API.Data.Models
{
    public class PostCommentVote
    {

        public int VoteId { get; set; }
        public int VoteCount { get; set; }

        public int CommentId { get; set; }

        public PostComment PostComment { get; set; }

        public ICollection<UserPostCommentVote> UserPostCommentVotes { get; set; }

        public PostCommentVote()
        {
            VoteCount = 0;
        }
    }
}
